src/BenchmarkDotNet.Core/Portability/HyperV.cs

```diff
@@ -0,0 +1,13 @@
+namespace BenchmarkDotNet.Portability
+{
+    public class HyperV : VirtualMachineHypervisor
+    {
+        public override string Name => "Hyper-V";
+
+        public override bool IsVirtualMachine(string manufacturer, string model)
+        {
+            return manufacturer != null && model != null && 
```

Comment:
I don't like that we have these null checks in each vm class. Could we do it once in `VirtualMachineHypervisor` and just call protected virtual method (which will be implemented in `HyperV`, `VirtualBox`, `VMware`) with the `([NotNull] string manufacturer, [NotNull] string model)` signature?